src/Symfony/Component/DependencyInjection/Secrets/JweEntry.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection\Secrets;
+
+
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+
+final class JweEntry
+{
+    //TODO do we need to store header and auth tag separately?
+    private $header;
+    private $encryptedKey;
+    private $iv;
+    private $cipherText;
+    private $authTag;
+
+    public static function decrypt(string $compactedEntry, string $keyPair)
+    {
+        $entry = new JweEntry();
+        $entry->hydrate($compactedEntry);
+        return $entry->decryptedSecret($keyPair);
+    }
+
+    public static function encrypt(string $secret, string $pubKey)
+    {
+        $entry = new JweEntry();
+        $entry->setEncryptedValues($secret, $pubKey);
+        return $entry->compact();
+    }
+
+    // see the JWE compact serialization format for details https://tools.ietf.org/html/rfc7516#section-7.1
+    protected function compact()
```

Comment:
Make everything private unless you actually need to sub-class it. Also, this class is final... so we definitely do not need to sub-class ;)

Reply:
👍 holdover from my ruby days, will do